File: saams-ui/src/model/company.ts
import { Base } from './base';
import Response from './response';

export interface Company extends Base {
    name: string;
    code: string;
    address: string;
    city: string;
    state: string;
    pincode: string;
    email: string;
    phone: string;
    fax: string;
}

export interface CompanyResponse extends Response {
    company: Company;
    companies: Company[];
}

File: Saams.Net/Saams.EF/UserManagement/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Saams.EF.UserManagement
{
    [Table("Companies")]
    public class Company : IEntity
    {
        public int Id { get; set; }

        [Required]
        public required string Name { get; set; }

        [Required]
        public required string Code { get; set; }

        [Required]
        public required string Address { get; set; }

        [Required]
        public required string City { get; set; }

        [Required]
        public required string State { get; set; }

        [Required]
        public required string Pincode { get; set; }

        [Required]
        public required string Email { get; set; }

        [Required]
        public required string Phone { get; set; }

        public string? Fax { get; set; }

        public ICollection<User> Users { get; set; } = new List<User>();
    }
}
